plugins/find/dialogs/find.js

```diff
@@ -486,8 +488,17 @@
 				if ( this.matchRange && this.matchRange.isMatched() && !this.matchRange._.isReplaced && !this.matchRange.isReadOnly() && !matchOptionsChanged ) {
 					// Turn off highlight for a while when saving snapshots.
 					this.matchRange.removeHighlight();
-					var domRange = this.matchRange.toDomRange();
-					var text = editor.document.createText( newString );
+					var domRange = this.matchRange.toDomRange(),
+						textWithPreservedSpaces = newString.replace( consecutiveWhitespaceRegex,
+							function( whitespace ) {
+								var newSpaces = CKEDITOR.tools.array.map( whitespace, function( space, i ) {
+									return i % 2 === 0 ? nonBreakingSpace : space;
+								} );
+
+								return newSpaces.join( '' );
+							} ),
```

Comment:
Overall code works good but let's move the replace logic into a separate function. It will look clearer.